src/Mvc/test/WebSites/RoutingWebSite/Areas/Admin/DynamicController.cs

```diff
@@ -0,0 +1,22 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNetCore.Mvc;
+
+namespace RoutingWebSite.Areas.Admin
+{
+    [Area("Admin")]
+    public class DynamicController : Controller
+    {
+        public ActionResult Index()
+        {
+            return Content("Hello from dynamic controller: " + Url.Action());
+        }
+
+        [HttpPost]
+        public ActionResult Index(int x = 0)
```

Comment:
Before this change, using a fallback route to map to this action would result in an implementation-defined choice of which method is called.